lib/middleman-images/image.rb

```diff
@@ -1,10 +1,48 @@
 module Middleman
   module Images
     class Image < Middleman::Sitemap::Resource
-      IGNORE_RESIZING = {
-        ".svg" => "WARNING: We did not resize %{file}. Resizing SVG files will lead to ImageMagick creating an SVG with an embedded binary image thus making the file way bigger.",
-        ".gif" => "WARNING: We did not resize %{file}. Resizing GIF files will remove the animation. If your GIF file is not animated, use JPG or PNG instead.",
+      IGNORE_ACTION = {
+        resize: {
+          ".svg" => "WARNING: We did not resize %{file}. Resizing SVG files will lead to ImageMagick creating an SVG with an embedded binary image thus making the file way bigger.",
+          ".gif" => "WARNING: We did not resize %{file}. Resizing GIF files will remove the animation. If your GIF file is not animated, use JPG or PNG instead.",
+        },
+        crop: {
+          ".svg" => "WARNING: We did not crop %{file}. Cropping SVG files will lead to ImageMagick creating an SVG with an embedded binary image thus making the file way bigger.",
+          ".gif" => "WARNING: We did not crop %{file}. Cropping GIF files will remove the animation. If your GIF file is not animated, use JPG or PNG instead.",
+        }
       }.freeze
+      MINI_MAGICK_OPTIONS = %i[adaptive-blur adaptive-resize adaptive-sharpen adjoin affine alpha annotate antialias
```

Comment:
Well this is a nightmare to maintain, but I do not see any alternative, if we would really try to support all options.

Reply:
I thought of this as well. The alternative could be to allow everything.
But this could lead to unexpected results if internal commands would be affected. Thus I kept the list for now.
Also if this is leading to many changes, it could be changed later on anyway.

Reply:
To avoid affecting internal commands, we could reserve a list of keywords. But in theory a whitelist is the nicer approach. I am torn, what do you think, @pmk1c ?

Reply:
I think we can keep the allowlist for now. If this really turns out to be hard to maintain, we can think about using a denylist for our internal options. But personally I think an allowlist is more explicit.

Reply:
(btw awesome wording Ruben :+1:)